src/client/renderer/r_state.c

```diff
@@ -391,41 +358,99 @@ void R_UploadToSubBuffer(r_buffer_t *buffer, const size_t start, const size_t si
 		// it's fairly rare you'll be editing at the end first,
 		// but who knows.
 		if (start) {
+
 			glBufferData(buffer->target, total_size, NULL, buffer->hint);
 			R_GetError("Partial resize");
+			r_view.num_buffer_full_uploads++;
 			glBufferSubData(buffer->target, start, size, data);
 			R_GetError("Partial update");
+			r_view.num_buffer_partial_uploads++;
 		} else {
 			glBufferData(buffer->target, total_size, data, buffer->hint);
 			R_GetError("Full resize");
+			r_view.num_buffer_full_uploads++;
 		}
 
+		r_state.buffers_total_bytes -= buffer->size;
+		r_state.buffers_total_bytes += total_size;
+
 		buffer->size = total_size;
 	} else {
 		// just update the range we specified
 		glBufferSubData(buffer->target, start, size, data);
-
 		R_GetError("Updating existing buffer");
+		r_view.num_buffer_partial_uploads++;
 	}
 
-	r_view.num_buffer_uploads++;
 	r_view.size_buffer_uploads += size;
 }
 
 /**
+ * @brief
+ */
+static GLubyte R_GetElementSize(const GLenum type) {
+
+	switch (type) {
+		case R_ATTRIB_BYTE:
+		case R_ATTRIB_UNSIGNED_BYTE:
+		case R_ATTRIB_INT_2_10_10_10_REV:
```

Comment:
I'm adding a note to this, but basically it's something of a hack. GL requires that INT_2_10_10_10_REV is defined as 4 bytes wide but also have a count of 4, even though the actual component is 4 bytes, not (4 * 4) bytes. Instead of having a completely separate if/else just to fix that one specific case where the GL spec is stupid as shit, I tell it that it's 1 byte so that the stride/size is correctly calculated as 4 bytes wide.